x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/mapper/SemanticTextInferenceResultFieldMapper.java

```diff
@@ -0,0 +1,241 @@
+ *         "_semantic_text_inference": {
+ *             "my_semantic_text_field": {
+ *                 [
+ *                     {
+ *                         "sparse_embedding": {
+ *                             "lucas": 0.05212344,
+ *                             "ty": 0.041213956,
+ *                             "dragon": 0.50991,
+ *                             "type": 0.23241979,
+ *                             "dr": 1.9312073,
+ *                             "##o": 0.2797593
+ *                         },
+ *                         "text": "these are not the droids you're looking for"
+ *                     }
+ *                 ]
+ *             }
+ *         }
+ *     }
+ * }
+ * </pre>
+ *
+ * This mapper parses the contents of the {@code _semantic_text_inference} field and indexes it as if the mapping were configured like so:
+ * <br>
+ * <br>
+ * <pre>
+ * {
+ *     "mappings": {
+ *         "properties": {
+ *             "my_semantic_text_field": {
+ *                 "type": "nested",
+ *                 "properties": {
+ *                     "sparse_embedding": {
+ *                         "type": "sparse_vector"
+ *                     },
+ *                     "text": {
+ *                         "type": "text",
+ *                         "index": false
+ *                     }
+ *                 }
+ *             }
+ *         }
+ *     }
+ * }
+ * </pre>
+ */
+public class SemanticTextInferenceResultFieldMapper extends MetadataFieldMapper {
+    public static final String CONTENT_TYPE = "_semantic_text_inference";
+    public static final String NAME = "_semantic_text_inference";
+    public static final String SPARSE_VECTOR_SUBFIELD_NAME = "sparse_embedding";
+    public static final String TEXT_SUBFIELD_NAME = "text";
+    public static final TypeParser PARSER = new FixedTypeParser(c -> new SemanticTextInferenceResultFieldMapper());
+
+    // TODO: Need to query this as a nested field type?
+    static class SemanticTextInferenceFieldType extends MappedFieldType {
+        private static final MappedFieldType INSTANCE = new SemanticTextInferenceFieldType();
+
+        SemanticTextInferenceFieldType() {
+            super(NAME, true, false, false, TextSearchInfo.NONE, Collections.emptyMap());
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
+            return SourceValueFetcher.identity(name(), context, format);
+        }
+
+        @Override
+        public Query termQuery(Object value, SearchExecutionContext context) {
+            return null;
+        }
+    }
+
+    private SemanticTextInferenceResultFieldMapper() {
+        super(SemanticTextInferenceFieldType.INSTANCE);
+    }
+
+    @Override
+    protected void parseCreateField(DocumentParserContext context) throws IOException {
```

Comment:
I chose to override `parseCreateField` (instead of `parse`) to take advantage of the error handling in the base class `parse` implementation.